src/common/src/util/memcmp_encoding.rs

```diff
@@ -430,14 +430,14 @@ mod tests {
     fn test_memcomparable_structs() {
         // NOTE: `NULL`s inside composite type values are always the largest.
 
-        let struct_none = None;
-        let struct_1 = Some(
+        let struct_none = Datum::None;
+        let struct_1 = Datum::Some(
```

Comment:
Why bother change this?

Reply:
Due to `auto_impl(&)`, it seems the type inference will fail as we use `.into()` in `Some(..)`. 🤣

Reply:
Actually I think `Datum::None`/`Datum::Some` can be a good practice, although changing one place doesn't make big difference in regard to maintainability.